setup.py

```diff
@@ -47,9 +47,9 @@
     cputune = ['-mcpu=native',]
 
 if os.name != 'nt':
-    compile_args = ['-O3', '-ffast-math', '-fno-associative-math']
+    compile_args = ['-std=c++11','-O3', '-ffast-math', '-fno-associative-math']
```

Comment:
what's the reason we need to add support for c++11?

Reply:
I got a lot of 'thread' is not a member of std and similar warnings for other <thread> functions. Would compiling for c++ 11 cause issues for *nix users?

Reply:
This is why I used pthreads in my branch, but if that doesn't work on Windows... is that a lose-lose then? What's the de-facto cross-platform threading?

Reply:
You could use boost::thread, maybe, since it's already using boost python?

Reply:
annoy doesn't use boost any more since it caused a lot of problems installing annoy, so i'm not too excited about reintroducing it

Reply:
could maybe just have some `#ifdef` magic and only support threading on posix systems?

Reply:
Is OpenMP an option?

Reply:
would avoid anything that require any linking with libraries just because it gets more complex to install

Reply:
well, then stick with c++11 then, since std::thread was standardized in c++11?

Reply:
ok seems like that's the logical conclusion :)